src/Compilers/CSharp/Portable/Syntax/InternalSyntax/TypeDeclarationSyntax.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using CoreSyntax = Microsoft.CodeAnalysis.Syntax.InternalSyntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
+{
+    internal abstract partial class TypeDeclarationSyntax
+    {
+        public abstract TypeDeclarationSyntax UpdateCore(
```

Comment:
Added as the internal syntax generator doesn't  include these abstract/overrides in that model.

Reply:
i really don't ove this.  but this mirrors how normal public syntax works.  And, importantly, means that if we ever add a new type-decl subclass, we'll know we have to ensuer that thsi works.